game/civ5map.ksy

```diff
@@ -0,0 +1,157 @@
+        type: b4
+      - id: version
+        type: b4
+        
+      - id: width
+        type: u4
+      - id: height
+        type: u4
+      - id: players
+        size: 1
+        
+      - id: misc_settings_head
+        type: b5
+      - id: random_goodies
+        type: b1
+      - id: random_resources
+        type: b1
+      - id: world_wrap
+        type: b1
+      - id: misc_settings_tail
+        size: 3
+        
+      - id: terrain_list_len
+        type: u4
+      - id: feature1_list_len
+        type: u4
+      - id: feature2_list_len
+        type: u4
+      - id: resource_list_len
+        type: u4
+        
+      - id: unknown
+        type: u4
+        
+      - id: map_name_len
+        type: u4
+      - id: map_description_len
+        type: u4
+        
+      - id: terrain_list
+        type: null_terminated_str
+        size: terrain_list_len
+      - id: feature1_list
+        type: null_terminated_str
+        size: feature1_list_len  
+      - id: feature2_list
+        type: null_terminated_str
+        size: feature2_list_len
+      - id: resource_list
+        type: null_terminated_str
+        size: resource_list_len
+        
+      - id: maybe_lua_params
+        doc: |
+          I don't know what's in here. It accompanies the lua stuff, so maybe
+          it defines options that passed into the lua file or something?
+        size: 36
+        if: version == 0xC
+      - id: lua_script_len
+        type: u4
+        if: version == 0xC
+      - id: lua_script
+        type: null_terminated_str
+        size: lua_script_len
+        if: version == 0xC
+        
+      - id: map_name
+        type: str
+        size: map_name_len
+      - id: map_description
+        type: str
+        size: map_description_len
+        
+      - id: string3_len
+        type: u4
+        if: version >= 0xB
+      - id: string3
+        type: str
+        size: string3_len
+        if: version >= 0xB
+        
+    types:
+      null_terminated_str:
+        seq:
+          - id: values
+            type: strz
+            repeat: eos
+
+  mapdata:
+    seq:
+      - id: plot_matrix
+        type: plot_row
+        repeat: expr
+        repeat-expr: _root.header.height
+  
+    types:
+      plot_row:
+        seq:
+          - id: plot_list
+            type: plot
+            repeat: expr
+            repeat-expr: _root.header.width
+      plot:
+        doc: |
+          whatever_type_id is an index into the corresponding list in 
+          the header. With the exception of terrain, it can be 0xFF,
+          which represents nonetype.
+        seq:
+          - id: terrain_type_id
+            size: 1
+          - id: resource_type_id
+            size: 1
+          - id: feature1_type_id
+            size: 1
+          - id: river
+            doc: 0 (none), non-0 may indicate a direction as well
+            size: 1
+          - id: elevation
+            doc: 0 (none), 1 (hill), 2 (mountain)
```

Comment:
Should be a `enum`.

Reply:
See 5a1f9e406ee42f0f7de30f6d4ebd3f701ae27f6e